django/core/serializers/base.py

```diff
@@ -82,8 +82,13 @@ def serialize(self, queryset, *, stream=None, fields=None, use_natural_foreign_k
             # Use the concrete parent class' _meta instead of the object's _meta
             # This is to avoid local_fields problems for proxy models. Refs #17717.
             concrete_model = obj._meta.concrete_model
+            # When using natural primary keys and multi-table inheritance,
+            # retrieve the pk field of the parent. That field must be
+            # serialized, otherwise deserialization isn't possible.
+            pk = self.use_natural_primary_keys and concrete_model._meta.pk
+            pk_parent = pk and pk.remote_field and pk.remote_field.parent_link and pk
```

Comment:
`pk` is twice. I think the last `and pk` can be removed.

Reply:
It's correct, but probably the code is too confusing. An alternate formulation is:
```
if self.use_natural_primary_keys:
    pk = concrete_model._meta.pk
    pk_parent = pk if pk.remote_field and pk.remote_field.parent_link else None
else:
    pk_parent = None
```

Reply:
Thanks. You're right too much `and` 😄. The second version is more readable for me.